sub-packages/bionemo-amplify/tests/bionemo/amplify/test_convert.py

```diff
@@ -0,0 +1,298 @@
+    )
+
+    nemo_model = nemo_config.configure_model(tokenizer).to("cuda").eval()
+
+    embedding_hook = ForwardHook(lambda inputs, outputs: outputs[0].transpose(0, 1))
+    nemo_model.embedding.register_forward_hook(embedding_hook)
+
+    query_post_rot_hook = ForwardHook(lambda inputs, outputs: inputs[0].transpose(0, 1).flatten(-2, -1))
+    nemo_model.encoder.layers[0].self_attention.core_attention.register_forward_hook(query_post_rot_hook)
+
+    key_post_rot_hook = ForwardHook(lambda inputs, outputs: inputs[1].transpose(0, 1).flatten(-2, -1))
+    nemo_model.encoder.layers[0].self_attention.core_attention.register_forward_hook(key_post_rot_hook)
+
+    value_post_rot_hook = ForwardHook(lambda inputs, outputs: inputs[2].transpose(0, 1).flatten(-2, -1))
+    nemo_model.encoder.layers[0].self_attention.core_attention.register_forward_hook(value_post_rot_hook)
+
+    attn_output_hook = ForwardHook(lambda inputs, outputs: outputs[0].transpose(0, 1))
+    nemo_model.encoder.layers[0].self_attention.core_attention.register_forward_hook(attn_output_hook)
+
+    attn_linear_output_hook = ForwardHook(lambda inputs, outputs: outputs[0].transpose(0, 1))
+    nemo_model.encoder.layers[0].self_attention.linear_proj.register_forward_hook(attn_linear_output_hook)
+
+    encoder_block_hooks = [
+        ForwardHook(lambda inputs, outputs: outputs[0].transpose(0, 1)) for _ in range(len(nemo_model.encoder.layers))
+    ]
+    for i, hook in enumerate(encoder_block_hooks):
+        nemo_model.encoder.layers[i].register_forward_hook(hook)
+
+    if dtype is torch.float16 or dtype is torch.bfloat16:
+        nemo_model = Float16Module(nemo_config, nemo_model)
+
+    nemo_output = nemo_model(input_ids, attention_mask)
+
+    nemo_logits = nemo_output["token_logits"].transpose(0, 1).contiguous()[..., : tokenizer.vocab_size]
+
+    return {
+        "embeddings": embedding_hook.data,
+        "query_post_rot": query_post_rot_hook.data,
+        "key_post_rot": key_post_rot_hook.data,
+        "value": value_post_rot_hook.data,
+        "attn_output": attn_output_hook.data,
+        "attn_linear_output": attn_linear_output_hook.data,
+        "encoder_block_outputs": [hook.data for hook in encoder_block_hooks],
+        "logits": nemo_logits,
+    }
+
+
+def test_convert_amplify_120M_smoke(tmp_path):
+    model_tag = "chandar-lab/AMPLIFY_120M"
+    module = biobert_lightning_module(config=AMPLIFYConfig())
+    io.import_ckpt(module, f"hf://{model_tag}", tmp_path / "nemo_checkpoint")
+
+
+def test_convert_amplify_120M(tmp_path):
+    model_tag = "chandar-lab/AMPLIFY_120M"
+    module = biobert_lightning_module(config=AMPLIFYConfig())
+    io.import_ckpt(module, f"hf://{model_tag}", tmp_path / "nemo_checkpoint")
+    with megatron_parallel_state_utils.distributed_model_parallel_state():
+        assert_amplify_equivalence(tmp_path / "nemo_checkpoint", model_tag, atol=1e-4, rtol=1e-4)
+
+
+def test_convert_amplify_120M_bf16(tmp_path):
```

Comment:
Do you also have a downstream accuracy test? Some kind of biological function interpretation or zero shot or something task? Those are important since sometimes you can have good cosine similarity and trash downstream accuracy or vice versa.

Reply:
we could calculate the loss for a real protein perhaps